### PR TITLE
fix: adding missing efi paths

### LINKED ISSUE #925: dracut-initramfs-restore creates non-chrootable environment
**Describe the bug**
/usr/lib/dracut/dracut-initramfs-restore script produces a environment, which cannot be chrooted into.

**Distribution used**
Arch Linux

**Dracut version**
50

**Init system**
systemd

**To Reproduce**
invoke the restore script (doesn't matter if invoked by systemd or manually)

    /usr/lib/dracut/dracut-initramfs-restore
    chroot /run/initramfs

output: chroot: failed to run command ‘/bin/bash’: No such file or directory

**Expected behavior**
If the chroot fails, the shutdown script will not clean up further after systemd. This leads to all kinds of follow up errors (lvm devices not properly disabled, all caches marked dirty etc.)

**Additional context**
Funnily the dracut script lsinitrd --extract does create an environment, which can be chrooted into. I have absolutely no clue, why.

I attach my dracut configs for some insight.

[myflags.conf.txt](https://github.com/dracutdevs/dracut/files/5221429/myflags.conf.txt)
[dm_caching.conf.txt](https://github.com/dracutdevs/dracut/files/5221432/dm_caching.conf.txt)
[nvidia.conf.txt](https://github.com/dracutdevs/dracut/files/5221433/nvidia.conf.txt)

### PR DESCRIPTION
**Pull request by johannbg**
This pull request adds missing efi paths.

### CODE REVIEW

File: dracut-initramfs-restore.sh [hunk 1]
```diff
@@ -6,20 +6,33 @@ set -e
...
 
 [[ $dracutbasedir ]] || dracutbasedir=/usr/lib/dracut
-SKIP="$dracutbasedir/skipcpio"
+
+if [[ -d "$dracutbasedir/skipcpio" ]]; then
+    SKIP="$dracutbasedir/skipcpio/skipcpio"
+    else
+    SKIP="$dracutbasedir/skipcpio"
+fi
+
```
Comment (haraldh) [inline dracut-initramfs-restore.sh:18]: These are unrelated changes... please remove and open a new PR for them

**Review by haraldh — CHANGES_REQUESTED** [review]

File: dracut-initramfs-restore.sh (hunk 1 above)
Comment (johannbg) [reply dracut-initramfs-restore.sh]: Removed

File: dracut-initramfs-restore.sh [hunk 2]
```diff
@@ -10,16 +10,24 @@ KERNEL_VERSION="$(uname -r)"
 
 [[ $dracutbasedir ]] || dracutbasedir=/usr/lib/dracut
 SKIP="$dracutbasedir/skipcpio"
+
```
Comment (haraldh) [inline dracut-initramfs-restore.sh:13]: ?

**Review by haraldh — CHANGES_REQUESTED** [review]

File: dracut-initramfs-restore.sh [hunk 3]
```diff
@@ -10,16 +10,24 @@ KERNEL_VERSION="$(uname -r)"
...
 
 mount -o ro /boot &>/dev/null || true
 
-if [[ $MACHINE_ID ]] && [[ -d /boot/${MACHINE_ID} || -L /boot/${MACHINE_ID} ]] ; then
-    IMG="/boot/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
+if [[ -d /efi/loader/entries || -L /efi/loader/entries ]] \
+    && [[ $MACHINE_ID ]] \
+    && [[ -d /efi/${MACHINE_ID} || -L /efi/${MACHINE_ID} ]] ; then
+        IMG="/efi/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
+    elif [[ -d /boot/loader/entries || -L /boot/loader/entries ]] \
```
Comment (haraldh) [inline dracut-initramfs-restore.sh:24]: please indent to start of line

File: dracut-initramfs-restore.sh [hunk 4]
```diff
@@ -10,16 +10,24 @@ KERNEL_VERSION="$(uname -r)"
...
+        && [[ $MACHINE_ID ]] \
+        && [[ -d /boot/${MACHINE_ID} || -L /boot/${MACHINE_ID} ]] ; then
+        IMG="/boot/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
+    else
```
Comment (haraldh) [inline dracut-initramfs-restore.sh:28]: please indent to start of line (same as `if` and `fi`) to be consistent with the rest of the code style

**Review by haraldh — CHANGES_REQUESTED** [review]

**Review by danimo — APPROVED** [review]

**Review by haraldh — APPROVED** [review]

**Pull request merged**